flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1801,7 +1847,7 @@ class ScalarExprLowering {
         caller.placeAddressAndLengthInput(arg, fir::getBase(argRef),
                                           fir::getLen(argRef));
       } else {
-        llvm_unreachable("pass by value not handled here");
+        TODO(loc, "pass by value in non elemental function call");
```

Comment:
Is there anything to be done here? Also, is this necessarily an elemental call?

Reply:
Yes, we need to ensure we are passing a temp (or that the callee is making a temp, I need to check what is the design here). I made a TODO because this is hit by programs like:

```
  interface
    subroutine foo(x)
      real, value :: x(100)
    end subroutine
  end interface
  real :: x(100) = 42.
  call foo(x)
end
```

> Also, is this necessarily an elemental call?

No on the contrary, you implemented VALUE for the elemental call case, this TODO is about the `non elemental` case.